### PR TITLE
Add settings to rbac python file.

### PR DESCRIPTION
**Pull request by myersCody**
## Jira Ticket

[COST-####](https://issues.redhat.com/browse/COST-####)

## Description

This change will ...

## Testing

1. Checkout Branch
2. Restart Koku
3. Hit endpoint or launch shell
    1. You should see ...
4. Do more things...

## Notes

...

### CODE REVIEW

**Review by maskarb — COMMENTED** [review]
This looks good, but we ought to add back the `if settings.ENHANCED_ORG_ADMIN and request.user.admin:` to the `SettingsAccessPermission` class.

File: koku/koku/test_rbac.py [hunk 1]
```diff
@@ -20,6 +20,28 @@
...
+        "gcp.project": default_read,
+        "openshift.cluster": default_read,
+        "openshift.node": default_read,
+        "openshift.project": default_read,
+        "ibm.account": default_read,
+        "oci.payer_tenant_id": default_read,
+    }
+    for key, value in access_dict.items():
+        default[key] = value
+    return default
```
Comment (samdoran) [inline koku/koku/test_rbac.py:42]: You can use the `update()` method to modify the values in `default` or the `|` operator to return a new dictionary with merged values. 
```suggestion
    return default | access_dict
```

File: koku/koku/test_rbac.py [hunk 2]
```diff
@@ -20,6 +20,28 @@
 from koku.rbac import RbacConnectionError
 from koku.rbac import RbacService
 
+
+def create_default_access(access_dict={}, default_write={"write": [], "read": []}, default_read={"read": []}):
+    """Helper function for defaulting access permissions."""
```
Comment (samdoran) [inline koku/koku/test_rbac.py:25]:
```suggestion
def create_default_access(access_dict=None, default_write=None):
    """Helper function for defaulting access permissions."""
    if access_dict is None:
        access_dict = {}
       
   if default_write is None:
       default_write = {"write": [], "read": []}, default_read={"read": []}
```

**Review by samdoran — CHANGES_REQUESTED** [review]

**Comment by maskarb** [comment]
> > This looks good, but we ought to add back the `if settings.ENHANCED_ORG_ADMIN and request.user.admin:` to the `SettingsAccessPermission` class.
> 
> I am pretty certain this would cause the current ddf settings to have a 403 again. The only reason that got resolved yesterday was because I removed the `settings.ENHANCED_ORG_ADMIN` check.
> 
> So, what I am trying to do is build settings permissions that will work for the current DDF settings & for the new settings endpoint.

I am 99% sure the reason we saw the 403 was because the `settings` in resource types was missing. An org admin has the `Cost Administrator` role automatically which includes the `cost-management:*:*` access. When we look up this access, we convert the wildcards [here](https://github.com/project-koku/koku/blob/main/koku/koku/rbac.py#L119) to give that user access to all the resource types. Since settings was missing, we were returning the 403.

**Review by maskarb — APPROVED** [review]

File: koku/koku/test_rbac.py [hunk 3]
```diff
@@ -20,6 +20,34 @@
...
 from koku.rbac import RbacService
 
+
+def create_expected_access(access_dict=None, default_write=None, default_read=None):
+    """Helper function for defaulting access permissions."""
+    if access_dict is None:
+        access_dict = {}
+
+    if default_write is None:
+        default_write = {"write": [], "read": []}
```
[the remark this answers is not in the document]
Comment (myersCody) [reply koku/koku/test_rbac.py:30]: So, I tried your suggestion of combining them, but the test starting failing. I need to be able to set them independently in certain places in the test:

https://github.com/project-koku/koku/pull/4586/files#diff-e99710acde7df448c3ce00e9fdf21f96a49c90549781c51d57f9a4292423df13R374

**Pull request merged**